### PR TITLE
Upgrade Radix UI to fix mismatch `id`

### PR DESCRIPTION
**Pull request by vitokhangnguyen**
Kudos to the [Radix](https://github.com/radix-ui) team for fixing this pretty quickly (relatively quickly since we only got the problem yesterday)

# Changelog:

* Upgrade Radix UI's Dropdown Menu nad React ID packages
* Remove custom `useUuid()` hook and use Radix's `useId()` instead in `@ui/Tab` and `@ui/Modal`.
* Remove `IdProvider` wrapper

### CODE REVIEW

File: src/ui/Modal/Modal.tsx [hunk 1]
```diff
@@ -48,7 +51,7 @@ const Modal: FC<ModalProps> = ({
             {closeButtonIcon ?? <Icon icon={closeFill} tw="w-6 h-6" />}
             <span tw="sr-only">Close dialog</span>
           </ButtonGhost>
-          <Text component="h3" variant="h5" id={`${id}-modal-title`}>
+          <Text component="h3" variant="h5" id={`Modal-Title-${modalId}`}>
```
Comment (Andrewnt219) [inline src/ui/Modal/Modal.tsx:54]: That's a weird naming convention. Pascal Kebab?

Comment (vitokhangnguyen) [reply src/ui/Modal/Modal.tsx:54]: Idk, never thought of it 😂

Switched to Pascal.

Comment (Andrewnt219) [reply src/ui/Modal/Modal.tsx:54]: This is worse... do it the same as classnames: lower-kebab-case.

Comment (Andrewnt219) [reply src/ui/Modal/Modal.tsx:54]: PascalCase is fine too.

Comment (Andrewnt219) [reply src/ui/Modal/Modal.tsx:54]: But not both.

image

Comment (vitokhangnguyen) [reply src/ui/Modal/Modal.tsx]: > This is worse... do it the same as classnames: lower-kebab-case.

Updated, generally I would have different convention between id and className but considering we don't use className much here, I can use the `lower-kebab-case`

**Review by Andrewnt219 — APPROVED** [review]

**Pull request merged**